### ISSUE TITLE
Image resizing 

### ISSUE THREAD

**Issue by rafaelzemog** [Witiko/markdown#50]
As discused in [#49](https://github.com/Witiko/markdown/issues/49) about the image example.

I have a lot of images in my markdown files but when I try to resize them the parser does nothing. Reading the documentation the parser only takes the name of the file and adjust the size to \textwidth. Is there any adittional option to include in order to resize any included image in markdown?
As far as I know an image can be resized with:
```
![example image](example-image.png "An example image from Martin Scharrer's mwe package" =250x250)
```

**Comment by Witiko** [comment]
Hello, as you have discovered, there is currently no support for image resizing, although you can [redefine the image renderer](http://mirrors.ctan.org/macros/generic/markdown/markdown.html#image-renderer) to make the label contain additional information, such as:

``` markdown
![250x250](example-image.png "An example image from Martin Scharrer's mwe package")
```

Note that TeX does not deal in pixels, so the size only makes sense if you assume some DPI (such as 72, which is used on the web), and use the assumption to convert pixels to centimeters and other units that TeX understands (250px at 72 DPI would be 3.47 inches or 8.81 cm).

There are many markdown syntax extensions that this package does not support, but I have not come across this syntax. Out of curiosity, which tool are you using that can resize images using this syntax?

**Comment by rafaelzemog** [comment]
My environment is Atom using the Markdown Preview package I also write in MacDown. My main OS is Mac OSX Mojave. Basically I just  moved from MS Office and made my own latex template fed by markdown files. After a few online research I found this instructions in order to adjust the maximum size of any imported image.
```
\makeatletter
\setkeys{Gin}{width=\ifdim\Gin@nat@width>\linewidth
    0.5\linewidth
\else
    \Gin@nat@width
\fi}
\makeatother
```
Thanks for your fast reply

**Comment by Witiko** [comment]
Thank you for sharing your setup and solution. The aim of the Markdown package is to allow the author to leverage both their knowledge of Markdown and their knowledge of LaTeX. I am looking forward to seeing your renderers.

As for the syntax extension, the Atom's Markdown Preview package [seems](https://github.com/atom/markdown-preview/blob/master/package.json#L16) to be using the [Marked](https://marked.js.org/) markdown parser. The documentation of Marked only claims the support of two Markdown flavors: [CommonMark](http://spec.commonmark.org/0.28/) and [GitHub Flavored Markdown](https://github.github.com/gfm/). To my knowledge, neither flavor supports the above syntax for resizing images. However, if we manage to track this extension down, I am in favor of adding this syntax extension to the Markdown package.

**Comment by Witiko** [comment]
Any news on this, @rafaelzemog?

**Comment by Migacz85** [comment]
@Witiko  Since we are using mwe, is not possible to just pass argument to mwe package so it will resize individual picture?

**Comment by Witiko** [comment]
@Migacz85 Definitely. Here is how we would do that after #123 has been implemented:

``` tex
\documentclass{article}
\usepackage[linkAttributes]{markdown}
\usepackage{graphicx, expl3}
\makeatletter
\ExplSyntaxOn
\markdownSetup{
  renderers = {
    linkAttributeContextBegin = {
      \group_begin:
      \markdownSetup{
        renderers = {
          attributeKeyValue = {
             \setkeys
               { Gin }
               { {#1} = {#2} }
          },
        },
      }
    },
    image = {
      \includegraphics
        { #2 }
    },
    linkAttributeContextEnd = {
      \group_end:
    },
  },
}
\ExplSyntaxOff
\makeatother
\begin{document}
\begin{markdown}
![example image](example-image "An example image from Martin Scharrer's mwe package"){width=1cm height=1cm}
\end{markdown}
\end{document}
```

**Comment by Migacz85** [comment]
@Witiko exactly that was what I was thinking about. 

Is that possible that this attributes can be extended? for example:

`scale=0.3` :

\begin{figure}
\centering
\includegraphics[scale=.3]{example-image}

`position=h`

\begin{figure}[h]
\includegraphics[width=8cm]{Plot}
\end{figure}

and support for the wrap would be awesome:

`wrap=true, position=r, scale=0.25: `


\begin{wrapfigure}{r}{0.25\textwidth} %this figure will be at the right
    \centering
    \includegraphics[width=0.25\textwidth]{mesh}
\end{wrapfigure}

**Comment by Witiko** [comment]
@Migacz85 Definitely, the above is just a minimal illustraction of the mechanism. You can explicitly react to different keys differently in the `attributeKeyValue` renderer above.

**Comment by Migacz85** [comment]
Hmmm I've tried it like that:

![image](https://user-images.githubusercontent.com/25789997/208254280-4e6124b7-35ae-4d1a-a7b6-448bd533a6e4.png)

But somehow text from markdown don't want to comply:

![image](https://user-images.githubusercontent.com/25789997/208254286-f45739fd-a4e9-4ec8-b3fd-48393ee4d765.png)

**Comment by Witiko** [comment]
@Migacz85 Not sure what kind of trick wrapfig uses that markdown breaks. Does it help to remove the `\end{markdown}` and `\begin{markdown}` around the wrapfig, or include the markdown text from an external file using `\markdownInput{asdf.md}`?

**Comment by Migacz85** [comment]
I should be more specific the last figure with correctly align picture is without tags `begin{markdown}, \end{markdown}` .  With `\markdownInput` - same effect.

**Comment by Witiko** [comment]
> I should be more specific the last figure with correctly align picture is without tags begin{markdown}, \\end{markdown} .

Here is what I meant:

![208254280-4e6124b7-35ae-4d1a-a7b6-448bd533a6e4.png](https://user-images.githubusercontent.com/603082/208265380-9dc08008-ec89-44c1-a2c3-4edf963db6dd.png)

Enabling the `hybrid` option should allow you to type LaTeX without dropping out of the markdown environment.

**Comment by Migacz85** [comment]
I just tested this, and unfortunately still doesn't work. I'm using hybrid
and when I pasted code for wrapping directly in Markdown the figure was on
the right, and the text was going on the figure.

**Comment by Witiko** [comment]
[The `wrapfig` documentations][1] says that you can specify the number of lines that will be pushed aside (here 20) using an optional parameter:

``` tex
\begin{wrapfigure}[20]{r}{0.4\textwidth}
```

I am not sure how `wrapfig` does its pushing-aside, but this may help.

 [1]: http://mirrors.ctan.org/macros/latex/contrib/wrapfig/wrapfig-doc.pdf
